src/main/scala/com/hungry/cars/http/in/CreateUserRequest.scala

```diff
@@ -0,0 +1,33 @@
+package com.hungry.cars.http.in
+
+import com.hungry.cars.domain.User
+import com.hungry.cars.domain.UserId
+import io.circe.Encoder
+import io.circe.Json
+import io.circe.syntax.EncoderOps
+
+case class CreateUserRequest(username: String, password: String, firstName: String, lastName: String, age: Int) {
+
+  def toUser: User = User(
```

Comment:
Tutaj dopuszczasz do sytaucji ze moze zostac utworzona instancja User'a, ktora zawiera nieprawidlowe dane wewnatrz. Po to walidujesz wszystko, zeby miec pewnosc ze taka sytuacja nie bedzie miala miejsca nigdzie w code-base'ie. `Make illegal state unrepresentable ~ Debashish Ghosh (Functional and Reactive domain modeling)`